src/publisher/index.ts

```diff
@@ -236,14 +254,28 @@ export class Publisher {
 
     this.settings = {
       batching: {
-        maxBytes: Math.min(batching.maxBytes, BATCH_LIMITS.maxBytes!),
-        maxMessages: Math.min(batching.maxMessages, BATCH_LIMITS.maxMessages!),
-        maxMilliseconds: batching.maxMilliseconds,
+        maxBytes: Math.min(batching!.maxBytes!, BATCH_LIMITS.maxBytes!),
+        maxMessages: Math.min(
+          batching!.maxMessages!,
```

Comment:
All the `!` makes me nervous we're committing TypeScript abuse - are these truly nullable?

Reply:
They're not in this particular case, unless `promisifyAll()` is breaking things for you (argh). It's using `extend()` to pull in a full tree of defaults. The original had the defaults inside the same method, so it was anonymously typed with non-nullable values.

Can we refer to these expressions where every value is followed by `!` as "the big bang"? >_>